controllers/mirrorpeer_controller.go

```diff
@@ -106,6 +108,19 @@ func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	logger.V(2).Info("All validations for MirrorPeer passed", "MirrorPeer", req.NamespacedName)
 
+	if mirrorPeer.Labels == nil {
+		mirrorPeer.Labels = make(map[string]string)
+	}
+
+	if val, ok := mirrorPeer.Labels[hubRecoveryLabel]; !ok || val != "resource" {
+		logger.Info("Adding label to mirrorpeer")
+		mirrorPeer.Labels[hubRecoveryLabel] = "resource"
+		err = r.Client.Update(ctx, &mirrorPeer)
```

Comment:
Once we update, status updates will fail because they will be operating on older version of resource. So we need to requeue when `err == nil`. So every object on first reconcile will be labelled and requeued.